api/task_processor/decorators.py

```diff
@@ -68,3 +67,42 @@ def run_in_thread(*, args: typing.Tuple = (), kwargs: typing.Dict = None):
         return f
 
     return decorator
+
+
+def register_recurring_task(
+    run_every: timedelta,
+    task_name: str = None,
+    args: typing.Tuple = (),
+    kwargs: typing.Dict = None,
+):
+    if not os.environ.get("RUN_BY_PROCESSOR"):
+        # Do not register recurring tasks if not running by task processor
+        return lambda f: f
+
+    def decorator(f: typing.Callable):
+        nonlocal task_name
+
+        task_name = task_name or f.__name__
+        task_module = getmodule(f).__name__.rsplit(".")[-1]
+        task_identifier = f"{task_module}.{task_name}"
+        register_task(task_identifier, f)
+        task = Task.schedule_task(
+            schedule_for=timezone.now(),
+            task_identifier=task_identifier,
+            run_every=run_every,
+            args=args,
+            kwargs=kwargs,
+        )
+        task, _ = Task.objects.update_or_create(
```

Comment:
Let's remove the call to `Task.schedule_task` and only have the `update_or_create` call. Let's also add a comment to explain why (because of the chance of multiple processor instances running).